src/function/scalar/string/like.cpp

```diff
@@ -1,40 +1,77 @@
-#include "duckdb/function/scalar/string_functions.hpp"
-
 #include "duckdb/common/exception.hpp"
 #include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/function/scalar/string_functions.hpp"
 
 using namespace std;
 
 namespace duckdb {
 
 static bool like_operator(const char *s, const char *pattern, const char *escape);
 
+struct LikeEscapeOperator {
+	template <class TA, class TB, class TC> inline bool operator()(TA left, TB right, TC esc) {
+		auto s = left.GetData();
+		auto pattern = right.GetData();
+		auto escape = esc.GetData();
+		return like_operator(s, pattern, escape);
+	}
+};
+
+struct NotLikeEscapeOperator {
+	template <class TA, class TB, class TC> inline bool operator()(TA left, TB right, TC esc) {
+		auto s = left.GetData();
+		auto pattern = right.GetData();
+		auto escape = esc.GetData();
+		return !like_operator(s, pattern, escape);
+	}
+};
+
 struct LikeOperator {
 	template <class TA, class TB, class TR> static inline TR Operation(TA left, TB right) {
-		return like_operator(left.GetData(), right.GetData(), nullptr);
+		auto s = left.GetData();
+		auto pattern = right.GetData();
+		auto escape = nullptr;
+		return like_operator(s, pattern, escape);
```

Comment:
Please return it to how it was ;) if you want more clarity, you can rename "left" and "right" as parameters (i.e. TA str, TB pattern)